File: sdk/ios/SsdidCore/Sources/SsdidCore/Domain/Transport/DTO/DeviceDtos.swift
import Foundation

/// Request/response DTOs for device pairing operations.

public struct PairingInitRequest: Codable {
    public let did: String
    public let challenge: String
    public let primaryKeyId: String

    enum CodingKeys: String, CodingKey {
        case did, challenge
        case primaryKeyId = "primary_key_id"
    }
}

public struct PairingInitResponse: Codable {
    public let pairingId: String

    enum CodingKeys: String, CodingKey {
        case pairingId = "pairing_id"
    }
}

public struct PairingJoinRequest: Codable {
    public let pairingId: String
    public let publicKey: String
    public let signedChallenge: String
    public let deviceName: String
    public let platform: String

    enum CodingKeys: String, CodingKey {
        case pairingId = "pairing_id"
        case publicKey = "public_key"
        case signedChallenge = "signed_challenge"
        case deviceName = "device_name"
        case platform
    }
}

public struct PairingJoinResponse: Codable {
    public let status: String
}

public struct PairingApproveRequest: Codable {
    public let did: String
    public let keyId: String
    public let signedApproval: String

    enum CodingKeys: String, CodingKey {
        case did
        case keyId = "key_id"
        case signedApproval = "signed_approval"
    }
}

public struct PairingStatusResponse: Codable {
    public let status: String
    public var deviceName: String? = nil
    public var publicKey: String? = nil
    public var signedChallenge: String? = nil

    enum CodingKeys: String, CodingKey {
        case status
        case deviceName = "device_name"
        case publicKey = "public_key"
        case signedChallenge = "signed_challenge"
    }
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/domain/transport/dto/DeviceDtos.kt
package my.ssdid.sdk.domain.transport.dto

import kotlinx.serialization.Serializable

@Serializable
data class PairingInitRequest(
    val did: String,
    val challenge: String,
    val primary_key_id: String
)

@Serializable
data class PairingInitResponse(val pairing_id: String)

@Serializable
data class PairingJoinRequest(
    val pairing_id: String,
    val public_key: String,
    val signed_challenge: String,
    val device_name: String,
    val platform: String
)

@Serializable
data class PairingJoinResponse(val status: String)

@Serializable
data class PairingApproveRequest(
    val did: String,
    val key_id: String,
    val signed_approval: String
)

@Serializable
data class PairingStatusResponse(
    val status: String,
    val device_name: String? = null,
    val public_key: String? = null,
    val signed_challenge: String? = null
)
